shoop/admin/modules/shops/views/edit.py

```diff
@@ -20,3 +24,14 @@ class ShopEditView(CreateOrUpdateView):
     form_class = modelform_factory(Shop, MultiLanguageModelForm, exclude=("owner", "options"))
     template_name = "shoop/admin/shops/edit.jinja"
     context_object_name = "shop"
+
+    def get_object(self, queryset=None):
+        if not settings.SHOOP_ENABLE_MULTIPLE_SHOPS and (Shop.objects.count() > 1 or not self.kwargs["pk"]):
+            raise Problem(_(u"Only one shop permitted."))
```

Comment:
I would split this if to two separate iffs from the "and" to ease the readability.